fil_fungible_token/src/token/mod.rs

```diff
@@ -19,43 +18,53 @@ pub enum TokenError {
     State(#[from] StateError),
     #[error("invalid negative: {0}")]
     InvalidNegative(String),
+    #[error("error calling receiver hook: {0}")]
+    MethodCall(#[from] MethodCallError),
+    #[error("receiver hook aborted when {from:?} sent {value:?} to {to:?} by {by:?}")]
+    ReceiverHook { from: ActorID, to: ActorID, by: ActorID, value: TokenAmount },
 }
 
 type Result<T> = std::result::Result<T, TokenError>;
 
 /// Library functions that implement core FRC-??? standards
 ///
 /// Holds injectable services to access/interface with IPLD/FVM layer.
-pub struct Token<BS>
+pub struct Token<BS, MC>
```

Comment:
Note: I don't know if this is a good suggestion, but an alternative here is to pull together all the external needs under a single trait, and then a single trait parameter here.